common/djangoapps/course_modes/views.py

```diff
@@ -135,6 +136,7 @@ def get(self, request, course_id, error=None):
 
         return render_to_response("course_modes/choose.html", context)
 
+    @method_decorator(transaction.non_atomic_requests)
```

Comment:
By default all views which are not decorated with `transaction.non_atomic_request` get wrapped in `transaction.atomic`, which is why we need to add this. See https://github.com/django/django/blob/1.8.5/django/core/handlers/base.py#L129-L132

Now that Django will not do any transaction management for this view, we wrap it `commit_on_success_with_read_committed` to setup our own custom transaction mangement.

Does that make sense?